### PR TITLE
Update VisualScriptCustomNode documentation

### LINKED ISSUE #50879: VisualScriptCustomNode working memory is not working
### Godot version

3.3.2

### System information

Windows 10

### Issue description

VisualScriptCustomNode
func _step(inputs, outputs, start_mode, working_mem):
working_mem is always `[]`

https://github.com/godotengine/godot/blob/438be4422f25ccc93aa47cd8ed9a01ff318949c7/modules/visual_script/visual_script_nodes.cpp#L2943

Between step and _step the working memory is swapped between `Variant *p_working_mem` and `Array work_mem`.
Now the `int work_mem_size` is never changed. So no size, no iteration, no memory.

When setting `work_mem_size = work_mem.size();` The project always fails on any line containing `p_working_mem[x]`.

If I can get some help converting from `Variant` to `Array` and back again? Then I think I can fix it.

### Steps to reproduce

Create a new custom node that has 2 sequence outputs and in _step append and print working memory

### Minimal reproduction project

Use Prototyping CustomNodeScript
[Autoload.zip](https://github.com/godotengine/godot/files/6877024/Autoload.zip)

Project
[VisualScriptCustomNodes.zip](https://github.com/godotengine/godot/files/6878020/VisualScriptCustomNodes.zip)

### PR DESCRIPTION
**Pull request by Gallilus**
It took me a long time to find what i was doing wrong. My thanks to MarianoRGD
Fix #50879
Cherry-pick 3.x

### CODE REVIEW

File: modules/visual_script/doc_classes/VisualScriptCustomNode.xml [hunk 1]
```diff
@@ -131,7 +131,7 @@
 				The [code]inputs[/code] array contains the values of the input ports.
 				[code]outputs[/code] is an array whose indices should be set to the respective outputs.
 				The [code]start_mode[/code] is usually [constant START_MODE_BEGIN_SEQUENCE], unless you have used the [code]STEP_*[/code] constants.
-				[code]working_mem[/code] is an array which can be used to persist information between runs of the custom node.
+				[code]working_mem[/code] is an array which can be used to persist information between runs of the custom node. The size needs to be predefined using [method _get_working_memory_size]. Do not overwrite [code]working_mem[/code] using a other [code]var[/code].
```
Comment (fire) [inline modules/visual_script/doc_classes/VisualScriptCustomNode.xml:134]: > using a other

Can you check the grammar and spelling?

Comment (Gallilus) [reply modules/visual_script/doc_classes/VisualScriptCustomNode.xml]: Beter?

Comment (fire) [reply modules/visual_script/doc_classes/VisualScriptCustomNode.xml]: > using a other

using another [code]var[/code].

Comment (Gallilus) [reply modules/visual_script/doc_classes/VisualScriptCustomNode.xml]: Ya, I installed Grammarly to help me.
image

File: modules/visual_script/doc_classes/VisualScriptCustomNode.xml [hunk 2]
```diff
@@ -131,7 +131,7 @@
 				The [code]inputs[/code] array contains the values of the input ports.
 				[code]outputs[/code] is an array whose indices should be set to the respective outputs.
 				The [code]start_mode[/code] is usually [constant START_MODE_BEGIN_SEQUENCE], unless you have used the [code]STEP_*[/code] constants.
-				[code]working_mem[/code] is an array which can be used to persist information between runs of the custom node.
+				[code]working_mem[/code] is an array which can be used to persist information between runs of the custom node. The size needs to be predefined using [method _get_working_memory_size]. Do not overwrite [code]working_mem[/code] using another [code]var[/code].
```
Comment (akien-mga) [inline modules/visual_script/doc_classes/VisualScriptCustomNode.xml:134]: > Do not overwrite [code]working_mem[/code] using another [code]var[/code].

That's a bit weird to add to the documentation. That's a very specific mention to a programming mistake which could have been done *anywhere* and is not related to `working_mem`. If you create a local variable that shadows a method parameter, then you're indeed taking the risk that accessing that identifier will maybe not access the instance you intended. I wouldn't call it out here for `working_mem` specifically, this should be part of the GDScript documentation (and might already be).

Comment (Gallilus) [reply modules/visual_script/doc_classes/VisualScriptCustomNode.xml]: Now the PR only ads " The size needs to be predefined using [method _get_working_memory_size]."

**Review by akien-mga — APPROVED** [review]
Looks good. Needs a rebase due to an unrelated GDScript unit test failure since fixed.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.4.

**Pull request merged**